src/rpc/coinjoin.cpp

```diff
@@ -341,7 +353,10 @@ static RPCHelpMan coinjoinsalt_set()
 
     const NodeContext& node = EnsureAnyNodeContext(request.context);
     if (node.coinjoin_loader != nullptr) {
-        auto cj_clientman = node.coinjoin_loader->walletman().Get(wallet->GetName());
+        auto cj_clientman = [&]() {
+            LOCK(node.coinjoin_loader->walletman().cs_wallet_manager_map);
```

Comment:
I'm starting to think it'd be a better idea to guard whaterver is needed inside of wallet man, and not expose the mutex publicly? Also we should consider changing the return types from being pointers which could dangle, to a shared_ptr maybe

Reply:
`Get()` is a `const` function, which forces locking the mutex outside the function body (unless we make `cs_wallet_manager_map` `mutable`), could probably make it a `shared_ptr` but so far it's fine and the current scope of the PR is to resolve problems with `develop`.

Reply:
yes we generally make mutexes mutable

Reply:
As for moving the mutex privately, `CoinJoinWalletManager::Add()` (which will require holding the mutex) will call `WalletInit::InitCoinJoinSettings()`, which will call `CoinJoinWalletManager::Get()` (which requires holding the mutex), this applies for `Add`, `Remove` and `Flush`. 

If we move the mutex privately, then there'd be more locking-unlocking compared to the current approach, where we lock the mutex in the interface (`CoinJoinLoaderImpl`), so when we call `AddWallet`, it's held, all operations (here, `Add` and `Get`) are completed, then released.

Reply:
How about ecf5c1ba76 instead of b89457dc9a0fcb6752d84649361fec5a6ad6d2e3?

Reply:
Resolved in latest push!